clients/src/test/java/org/apache/kafka/common/header/internals/RecordHeadersTest.java

```diff
@@ -220,9 +220,7 @@ public void shouldThrowNpeWhenAddingCollectionWithNullHeader() {
 
     private int getCount(Headers headers) {
         int count = 0;
-        Iterator<Header> headerIterator = headers.iterator();
-        while (headerIterator.hasNext()) {
-            headerIterator.next();
+        for (Header ignore : headers) {
```

Comment:
How about using `toArray`? for example: `headers.toArray().length`

Reply:
Nice suggestion